pulsar-client-cpp/lib/ConsumerImpl.cc

```diff
@@ -110,6 +110,10 @@ Future<Result, ConsumerImplBaseWeakPtr> ConsumerImpl::getConsumerCreatedFuture()
     return consumerCreatedPromise_.getFuture();
 }
 
+void ConsumerImpl::incrRefCount() { ++refCount_; }
+
+bool ConsumerImpl::shouldTearDown() { return refCount_ > 0 ? refCount_-- == 0 : refCount_ == 0; }
```

Comment:
the name here doesn't indicate it is decrementing the `refCount_`. I would suggest calling it something like `safeDecRefCount` and return the value of `refCount_` after decrementing.

Reply:
Great! I agree that it should express the fact that is being decremented.
I did it they way you mention If I understood correctly, please take a look now :+1: